From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mio <mio@mio19.uk>
Date: Fri, 23 Oct 2026 20:00:39 +1100
Subject: [PATCH] examples/asteroid: enable polkit for logind linger

The canonical linger-users.service calls loginctl enable-linger, but logind's SetUserLinger is denied through polkit on this image. Enable polkit explicitly alongside the set-user-linger rule so the standard linger path can create the marker and start user@1000.
---
 examples/asteroid/asteroid-desktop.nix | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/examples/asteroid/asteroid-desktop.nix b/examples/asteroid/asteroid-desktop.nix
index 1b32a327..8b2a81b7 100644
--- a/examples/asteroid/asteroid-desktop.nix
+++ b/examples/asteroid/asteroid-desktop.nix
@@ -196,6 +196,9 @@ in
       group = "root";
     };
 
+    # logind authorizes SetUserLinger through polkit.
+    security.polkit.enable = true;
+
     # Upstream AsteroidOS allows the ceres user to reboot and power off.
     # Reference: https://github.com/AsteroidOS/meta-asteroid/blob/master/recipes-asteroid/asteroid-apps/files/30-org.freedesktop.login1.rules
     security.polkit.extraConfig = ''
